### PR TITLE
Bash completion updates

### PR DESCRIPTION
**Pull request by nak3**
This patch set includes:
- `kubectl edit <tab>` gets resources and `kubectl edit RESOURCE <tab>` gets handled resources
- `kubectl patch <tab>` gets resources and `kubectl patch RESOURCE <tab>` gets handled resources
- `kubectl annotate <tab>` gets resources and `kubectl annotate RESOURCE <tab>` gets handled resources
- `kubectl expose <tab>` gets resources and `kubectl expose RESOURCE <tab>` gets handled resources

### CODE REVIEW

**Comment by sttts** [comment]
@nak3 please rebase onto https://github.com/kubernetes/kubernetes/pull/23262

File: contrib/completions/bash/kubectl [hunk 1]
```diff
@@ -2302,6 +2358,18 @@ _kubectl_annotate()
...
     must_have_one_noun=()
+    must_have_one_noun+=("componentstatuses")
+    must_have_one_noun+=("events")
+    must_have_one_noun+=("horizontalpodautoscalers")
+    must_have_one_noun+=("limitranges")
+    must_have_one_noun+=("nodes")
+    must_have_one_noun+=("persistentvolumeclaims")
+    must_have_one_noun+=("persistentvolumes")
+    must_have_one_noun+=("pods")
+    must_have_one_noun+=("replicationcontrollers")
```
Comment (sttts) [inline contrib/completions/bash/kubectl]: Wondering why these are all in plural form.

Comment (sttts) [inline contrib/completions/bash/kubectl]: Background: there is an `argAliases` field no in the Command, next to `validArgs`. That's meant to include the plural while the `validArgs` are singular.

**Comment by nak3** [comment]
@sttts OK, I will wait that your PR will be merged.

File: contrib/completions/bash/kubectl (hunk 1 above)
Comment (jlowdermilk) [inline contrib/completions/bash/kubectl]: cc @eparis

**Comment by eparis** [comment]
ok to test

File: docs/user-guide/kubectl/kubectl_annotate.md [hunk 2]
```diff
@@ -18,11 +18,6 @@
 If you are using a released version of Kubernetes, you should
 refer to the docs that go with that version.
 
-<!-- TAG RELEASE_LINK, added by the munger automatically -->
-<strong>
-The latest release of this document can be found
-[here](http://releases.k8s.io/release-1.2/docs/user-guide/kubectl/kubectl_annotate.md).
-
```
Comment (eparis) [inline docs/user-guide/kubectl/kubectl_annotate.md]: This seems odd, but we'll see what the CI system says.

File: contrib/completions/bash/kubectl (hunk 1 above)
Comment (eparis) [inline contrib/completions/bash/kubectl]: agree with @stts we generally prefer the singular in the bash completions.

**Comment by jlowdermilk** [comment]
@k8s-bot unit test this issue: #IGNORE

**Comment by nak3** [comment]
Thank you @sttts @eparis I updated.

**Comment by nak3** [comment]
@jlowdermilk sorry I dropped lgtm lable

**Comment by nak3** [comment]
rebased

**Comment by sttts** [comment]
@nak3 did you rerun genbashcomp?

**Comment by nak3** [comment]
@sttts Yes, I ran `hack/update-generated-docs.sh` and `genbashcomp` both. Did you find any issue?

**Comment by sttts** [comment]
@nak3 the `argAliases` field of the "new" commands is not set. Compare e.g. https://github.com/nak3/kubernetes/blob/bash-completion/pkg/kubectl/cmd/get.go#L77.

**Comment by nak3** [comment]
@sttts Oh, I understand. Thank you.

**Pull request merged**